end_to_end_testing/run_end_to_end.sh

```diff
@@ -57,7 +85,30 @@ function helm_release_has_key_value() {
     local key="${2}"
     local value="${3}"
 
-    if helm get values "${release_name}" --output json | jq -e ".[\"${key}\"] == \"${value}\""; then
+    if helm get values "${release_name}" --output json | jq -e ".[\"${key}\"] == \"${value}\"" &>/dev/null; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function helm_release_values_has_key() {
+    local release_name="${1}"
+    local key="${2}"
+
+    if helm get values "${release_name}" --output json | jq -e ". | has(\"${key}\")" &>/dev/null; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function helm_release_key_value_is_type() {
```

Comment:
It looks like this function is being called inside a `for` loop, resulting in many separate, identical calls to `helm get values`. Would it make more sense to grab that output once and then pass that to `value_is_type` and `values_has_key`?

Edit: realizing this is just a test file, so NBD

Reply:
We could definitely rely on a cached version of the objects captured after each reckoner plot command, although bash has it's limitations with variable size and string manipulation. Generally when stuff like this gets involved, it means we should probably move away from bash. I'm happy to chat about this more, but if this isn't blocking the PR then we maybe could open an issue.

Reply:
Given that it's just a test, I'd say leave it. I originally thought this was running on (potentially large) user inputs.